### PR TITLE
Add cloid to order and cancel requests

### PR DESCRIPTION
**Pull request by Nayshins**
Tested using the updated order and cancel script. I think the cloid was causing signing issues with the request in #6, and this should allow us to add the modify requests.

### CODE REVIEW

File: src/bin/order_and_cancel.rs [hunk 1]
```diff
@@ -55,4 +57,33 @@ async fn main() {
     // This response will return an error if order was filled (since you can't cancel a filled order), otherwise it will cancel the order
     let response = exchange_client.cancel(cancel, None).await.unwrap();
     info!("Order potentially cancelled: {response:?}");
+
+    // Order and Cancel with cloid
+    let cloid = Uuid::new_v4();
```
Comment (lmlmt) [inline src/bin/order_and_cancel.rs:62]: nit: could you separate the cloid example into a separate file, e.g. `order_and_cancel_with_cloid.rs`. This follows the pattern we have for the python-sdk, i.e. `basic_order_with_cloid.py`, in https://github.com/hyperliquid-dex/hyperliquid-python-sdk/tree/master/examples

**Review by lmlmt — COMMENTED** [review]
Thanks for submitting this! We are actually in the midst of updating our hashing scheme, which will especially simplify how the cloids are handled and prevent the issues we saw with an earlier version of this. We aim to get that in soon, along with an update to the SDK to switch to the new scheme. It's probably better to wait until those pieces are in, then adapting these actions to that scheme in this PR before merging it in.

**Comment by lmlmt** [comment]
FYI new hashing is merged

File: src/bin/order_and_cancel.rs (hunk 1 above)
Comment (Nayshins) [reply src/bin/order_and_cancel.rs:62]: can do

**Comment by Nayshins** [comment]
@lmlmt Updated to use the new signing. Massively cleaner impl

**Comment by dennohpeter** [comment]
Some suggestions on ```cloid```,  ```cloid``` can be of type `H128` from ethers, removing the need for an extra ```uuid``` dependency. To generate a new cloud is pretty simple: [H128::random()](https://docs.rs/ethers/latest/ethers/core/types/struct.H128.html)

**Comment by Nayshins** [comment]
Screenshot 2024-02-19 at 1 06 18 PM
Screenshot 2024-02-19 at 1 04 36 PM
Gonna stick with UUID IMO

**Comment by Nayshins** [comment]
Bench Results:
```
uuid v4                 time:   [512.99 ns 533.09 ns 553.06 ns]
Found 2 outliers among 100 measurements (2.00%)
  1 (1.00%) high mild
  1 (1.00%) high severe

H128::random            time:   [4.9576 µs 5.0533 µs 5.1658 µs]
Found 8 outliers among 100 measurements (8.00%)
  2 (2.00%) high mild
  6 (6.00%) high severe
 ```

**Comment by lmlmt** [comment]
> Some suggestions on `cloid`, `cloid` can be of type `H128` from ethers, removing the need for an extra `uuid` dependency. To generate a new cloud is pretty simple: [H128::random()](https://docs.rs/ethers/latest/ethers/core/types/struct.H128.html)

Yeah good idea, and also based on the bench results; feel free to push latest changes here and will take a look

**Comment by Nayshins** [comment]
Bench shows that UUID is 10x faster than H128 500ns vs 5 micros

**Comment by Nayshins** [comment]
> > Some suggestions on `cloid`, `cloid` can be of type `H128` from ethers, removing the need for an extra `uuid` dependency. To generate a new cloud is pretty simple: [H128::random()](https://docs.rs/ethers/latest/ethers/core/types/struct.H128.html)
> 
> Yeah good idea, and also based on the bench results; feel free to push latest changes here and will take a look

Bench shows that UUID is 10x faster than H128 500ns vs 5 micros. I think what I have now works the best IMO.

**Comment by lmlmt** [comment]
> > > Some suggestions on `cloid`, `cloid` can be of type `H128` from ethers, removing the need for an extra `uuid` dependency. To generate a new cloud is pretty simple: [H128::random()](https://docs.rs/ethers/latest/ethers/core/types/struct.H128.html)
> > 
> > 
> > Yeah good idea, and also based on the bench results; feel free to push latest changes here and will take a look
> 
> Bench shows that UUID is 10x faster than H128 500ns vs 5 micros. I think what I have now works the best IMO.

Ah, misread, will review

**Comment by lmlmt** [comment]
Also confirming you ran the other examples locally and ensured they worked, ie no signature errors?

**Comment by Nayshins** [comment]
Ran `order_and_cancel`, `order_and_cancel_cloid`, and `market_maker` without issue

File: src/exchange/mod.rs [hunk 2]
```diff
@@ -6,6 +6,7 @@ mod order;
 
 pub use actions::*;
 pub use cancel::ClientCancelRequest;
```
Comment (lmlmt) [inline src/exchange/mod.rs:8]: nit: group imports `pub use cancel::{ClientCancelRequest, ClientCancelRequestCloid};`

**Review by lmlmt — APPROVED** [review]
Looks good overall. I'm also adding more unit tests in a separate PR so feel free to add similar ones for cloid: https://github.com/hyperliquid-dex/hyperliquid-rust-sdk/pull/13

**Comment by dennohpeter** [comment]
> Bench Results:
> 
> ```
> uuid v4                 time:   [512.99 ns 533.09 ns 553.06 ns]
> Found 2 outliers among 100 measurements (2.00%)
>   1 (1.00%) high mild
>   1 (1.00%) high severe
> 
> H128::random            time:   [4.9576 µs 5.0533 µs 5.1658 µs]
> Found 8 outliers among 100 measurements (8.00%)
>   2 (2.00%) high mild
>   6 (6.00%) high severe
> ```

Yeah ```uuid v4``` is definitely more faster

**Comment by dennohpeter** [comment]
I have also opened a PR that improves your PR's internal performance when converting `uuid` to hex.  All the tests and function signatures remain the same but with ~2.3x speed boost. Have a look #14

**Comment by Nayshins** [comment]
Smart! I just left it as the original implementation I had to deal with using the original hashing scheme.

**Pull request merged**